File: math_ops.R
You said:
# math_ops.R
# Looping interactive calculator in R
# Exit only when user types 'q' or 'Q'

repeat {
  # Prompt for numbers
  num1 <- as.numeric(readline(prompt = "Enter first number (or q to quit): "))
  if (is.na(num1)) {
    input <- readline(prompt = "Did you mean to quit? (q to exit, any other key to continue): ")
    if (tolower(input) == "q") break else next
  }
  
  num2 <- as.numeric(readline(prompt = "Enter second number: "))
  
  # Show menu
  cat("\nChoose an operation:\n")
  cat("1: Addition\n")
  cat("2: Subtraction\n")
  cat("3: Multiplication\n")
  cat("4: Division\n")
  cat("5: Modulus (Remainder)\n")
  cat("q: Quit\n")
  
  choice <- readline(prompt = "Enter choice (1-5 or q): ")
  
  if (tolower(choice) == "q") {
    cat("\nExiting calculator. Goodbye!\n")
    break
  }
  
  choice <- as.integer(choice)
  
  # Perform operation
  result <- switch(choice,
                   num1 + num2,
                   num1 - num2,
                   num1 * num2,
                   if (num2 != 0) num1 / num2 else "Error: Division by zero",
                   if (num2 != 0) num1 %% num2 else "Error: Modulus by zero",
                   "Invalid choice")
  
  cat("\nResult: ", result, "\n\n")
}

File: math_ops.py
# math_ops.py
# Looping interactive calculator in Python
# Exit only when user types 'q' or 'Q'

while True:
    # Prompt for first number
    num1_input = input("Enter first number (or q to quit): ")
    
    if num1_input.lower() == 'q':
        print("\nExiting calculator. Goodbye!")
        break

    # Validate num1
    try:
        num1 = float(num1_input)
    except ValueError:
        confirm = input("Did you mean to quit? (q to exit, any other key to continue): ")
        if confirm.lower() == 'q':
            print("\nExiting calculator. Goodbye!")
            break
        else:
            continue

    # Prompt for second number
    num2 = float(input("Enter second number: "))

    # Show menu
    print("\nChoose an operation:")
    print("1: Addition")
    print("2: Subtraction")
    print("3: Multiplication")
    print("4: Division")
    print("5: Modulus (Remainder)")
    print("q: Quit")

    choice = input("Enter choice (1-5 or q): ")

    if choice.lower() == 'q':
        print("\nExiting calculator. Goodbye!")
        break

    # Perform operation
    result = None

    try:
        choice = int(choice)
        if choice == 1:
            result = num1 + num2
        elif choice == 2:
            result = num1 - num2
        elif choice == 3:
            result = num1 * num2
        elif choice == 4:
            result = num1 / num2 if num2 != 0 else "Error: Division by zero"
        elif choice == 5:
            result = num1 % num2 if num2 != 0 else "Error: Modulus by zero"
        else:
            result = "Invalid choice"
    except ValueError:
        result = "Invalid choice"

    print("\nResult:", result, "\n")